lib/TH/generic/THStorage.h

```diff
@@ -28,6 +29,8 @@ typedef struct THStorage
     char flag;
     THAllocator *allocator;
     void *allocatorContext;
+    int type;
```

Comment:
When we pass this storage data to an underlying transport mechanism (MPI or Libfabric) in a multi-machine environment it needs to know the type of data being transferred. For example if we pass a storage of floats from a little endian to a big endian machine the transport would re-order the bytes of each float for us.

We could maybe do it without storage->type but it would require us knowing machine endian-ness at some other level of the stack, not that easy to do.